source-code/plugins/i18next/src/plugin.ts

```diff
@@ -134,78 +158,101 @@ async function readResources(
-						continue
-					}
-					const file = (await args.$fs.readFile(`${directoryPath}/${potentialResourcePath}`, {
-						encoding: "utf-8",
-					})) as string
-
-					FILE_HAS_NEW_LINE[`${directoryPath}/${potentialResourcePath}`] = file.endsWith("\n")
-					SPACING[`${directoryPath}/${potentialResourcePath}`] = detectJsonSpacing(file)
+		const resource: ast.Resource = {
+			type: "Resource",
+			languageTag: {
+				type: "LanguageTag",
+				name: language,
+			},
+			body: [],
+		}
 
-					serializedMessages = [
-						...serializedMessages,
-						...collectNestedSerializedMessages(
-							JSON.parse(file),
-							[],
-							potentialResourcePath.replace(".json", ""),
+		if (typeof args.settings.pathPattern !== "string") {
+			for (const [prefix, path] of Object.entries(args.settings.pathPattern)) {
+				const messages = await getFileToParse(path, language, args.$fs)
+				if (messages) {
+					resource.body = [
+						...resource.body,
+						...parseBody(
+							messages,
+							args.settings.variableReferencePattern,
+							NESTED[path.replace("{language}", language)] ?? defaultNesting(),
+							prefix,
 						),
 					]
 				}
-			} else {
-				const file = (await args.$fs.readFile(resourcePath, {
-					encoding: "utf-8",
-				})) as string
+			}
+		} else {
+			const messages = await getFileToParse(args.settings.pathPattern, language, args.$fs)
+			resource.body = [
+				...resource.body,
+				...parseBody(
+					messages,
+					args.settings.variableReferencePattern,
+					NESTED[args.settings.pathPattern.replace("{language}", language)] ?? defaultNesting(),
+				),
+			]
+		}
+		resources.push(resource)
+	}
 
-				FILE_HAS_NEW_LINE[`${resourcePath}`] = file.endsWith("\n")
-				SPACING[`${resourcePath}`] = detectJsonSpacing(file)
+	return resources
+}
 
-				serializedMessages = collectNestedSerializedMessages(JSON.parse(file))
-			}
-			result.push(
-				parseResource(serializedMessages, language, args.settings.variableReferencePattern),
-			)
-		} catch (e) {
-			if ((e as any).code === "ENOENT") {
-				// file does not exist yet
-				continue
-			}
-			throw e
+/**
+ * Get the files that needs to be parsed.
+ */
+async function getFileToParse(path: string, language: string, $fs: InlangEnvironment["$fs"]) {
+	const pathWithLanguage = path.replace("{language}", language)
+	// get file, make sure that is not braking when the namespace doesn't exist in every language dir
+	try {
+		const file = await $fs.readFile(pathWithLanguage, { encoding: "utf-8" })
+		//analyse format of file
+		SPACING[pathWithLanguage] = detectJsonSpacing(file as string)
+		NESTED[pathWithLanguage] = detectIsNested(file as string)
+		FILE_HAS_NEW_LINE[pathWithLanguage] = (file as string).endsWith("\n")
+
+		return JSON.parse(file as string)
+	} catch (e) {
+		// if the namespace doesn't exist for this dir -> continue
+		if ((e as any).code === "FileNotFound" || (e as any).code === "ENOENT") {
+			// file does not exist yet
+			return
 		}
+		throw e
 	}
```

Comment:
Haha I was thinking about that. I simple couldn't find a error case that I could trigger in the test (easily), that would have thrown an error here. But I think just in case our file system throws another error we have it covered with the throw. I would leave it like that.